File: app/src/main/java/com/chinachino/mvvm/ViewModels/MovieDetailsViewModel.java
package com.chinachino.mvvm.ViewModels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.chinachino.mvvm.Repositories.MovieDetailsRepository;
import com.chinachino.mvvm.Repositories.MovieRepository;
import com.chinachino.mvvm.models.Details;

public class MovieDetailsViewModel extends ViewModel {
    private MovieDetailsRepository movieRepository;

    public MovieDetailsViewModel() {
        movieRepository = MovieDetailsRepository.getInstance();
    }
    public LiveData<Details> getMovieDetails(){
        return movieRepository.getMovieDetails();
    }
    public void SearchMovieDetails(int movieID){
        movieRepository.SearchMovieDetails(movieID);
    }
}

File: app/src/main/java/com/chinachino/mvvm/viewModels/MovieDetailsViewModel.kt
package com.chinachino.mvvm.ViewModels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.chinachino.mvvm.models.Details
import com.chinachino.mvvm.repositories.MovieDetailsRepository

class MovieDetailsViewModel : ViewModel() {
    var isMovieRetrieved = false
    private val movieRepository: MovieDetailsRepository = MovieDetailsRepository.instance!!
    val movieDetails: LiveData<Details?>
        get() = movieRepository.movieDetails

    fun searchMovieDetails(movieID: Int) {
        movieRepository.searchMovieDetails(movieID)
    }

    val isRequestTimedOut: LiveData<Boolean>
        get() = movieRepository.isRequestTimeOut

}